### PR TITLE
week-4-d

### PR DESCRIPTION
**Pull request by p8410077**

### CODE REVIEW

**Comment by jskzhou** [comment]
There are a significant number of typos and grammatical errors, which decreases the readability of the notes and decreases the strength and quality of content you have. A thorough proof-reading and spell-check can help the content of your notes come through much stronger and cleaner, and improve the experience for the reader of the notes.

File: bite-size-notes/week-4-d-Example of LQR-name.md [hunk 1]
```diff
@@ -0,0 +1,350 @@
+# Objectives: 
+1. Motivation of LQR
+2. Summarize the variation of LQR and corresponding Recati equation and optimal control. 
+3. Using ode45 to solve Reccati equation.
+# Introduction: 
+LQR (Linear Quatratic Regulater) is the a optimal solution to a full state feedback control to minimine the state error and control efforts. Let's break down this sentence to understand what is LQR ?
```
Comment (jskzhou) [inline bite-size-notes/week-4-d-Example of LQR-name.md]: I don't think this is true, LQR does not guarantee optimality for full state, much like PID and other weight tuning based methods. 

LQR _could_ yield optimal control inputs for the problem depending on the chosen weights, and provides the optimal solution for some cost function which is based on a chosen set of weights. That is not equivalent to the optimal solution to full-state feedback control. I feel that your statement is too strong, especially without proof for the general optimality that your statement implies.

File: bite-size-notes/week-4-d-Example of LQR-name.md [hunk 2]
```diff
@@ -0,0 +1,350 @@
+# Objectives: 
+1. Motivation of LQR
+2. Summarize the variation of LQR and corresponding Recati equation and optimal control. 
+3. Using ode45 to solve Reccati equation.
+# Introduction: 
+LQR (Linear Quatratic Regulater) is the a optimal solution to a full state feedback control to minimine the state error and control efforts. Let's break down this sentence to understand what is LQR ? 
+## What is a fullstate feedback control.
+We want to stabilize a system and drive all state to 0. A standard full state feed back control can achieve this goal.(There are other variation which are out of the scope) \
```
Comment (jskzhou) [inline bite-size-notes/week-4-d-Example of LQR-name.md]: I'm not sure what you mean here by "standard full-state feedback control", as there are many ways of dealing with feedback control.

File: bite-size-notes/week-4-d-Example of LQR-name.md [hunk 3]
```diff
@@ -0,0 +1,350 @@
...
+A standar full feedback controle has the block diagram. 
+ ![fullfeedback](https://github.com/p8410077/AA548-spr2024/assets/11802603/bea8ed3a-5062-4d60-93c9-b213d4d856d5)
+
+By using the full state control, we can using the method called pole placement, literally, we are putting the poles of the system at the desired location on the complex plane.
```
Comment (jskzhou) [inline bite-size-notes/week-4-d-Example of LQR-name.md]: Could elaborate a little on what is meant by full-state control, it's unclear how full-state control and pole placement relate to each other. Also, transitioning from plural "poles" to singular "location" can be confusing and implies that all poles are placed in the exact same position in the complex plane.

File: bite-size-notes/week-4-d-Example of LQR-name.md [hunk 4]
```diff
@@ -0,0 +1,350 @@
+# Objectives: 
+1. Motivation of LQR
+2. Summarize the variation of LQR and corresponding Recati equation and optimal control. 
+3. Using ode45 to solve Reccati equation.
+# Introduction: 
+LQR (Linear Quatratic Regulater) is the a optimal solution to a full state feedback control to minimine the state error and control efforts. Let's break down this sentence to understand what is LQR ? 
+## What is a fullstate feedback control.
+We want to stabilize a system and drive all state to 0. A standard full state feed back control can achieve this goal.(There are other variation which are out of the scope) \
+A standar full feedback controle has the block diagram. 
+ ![fullfeedback](https://github.com/p8410077/AA548-spr2024/assets/11802603/bea8ed3a-5062-4d60-93c9-b213d4d856d5)
```
Comment (jskzhou) [inline bite-size-notes/week-4-d-Example of LQR-name.md]: A diagram here is helpful but this diagram is a little confusing. The feedback gain can be clearer on the bottom of the loop. Explicitly showing input and output and using summing blocks and branches can help as well with clarity. Also, showing an error term somewhere can illustrate the purpose of feedback.

I find the diagram shown in this link to be intuitive to understand: https://www.electronics-tutorials.ws/systems/closed-loop-system.html

File: bite-size-notes/week-4-d-Example of LQR-name.md [hunk 5]
```diff
@@ -0,0 +1,350 @@
...
+
+$$
+y(t) = (C-DK)x(t)
+$$
+
+We know that the eigenvalue of the matrix (A-BK) is the poles of the system. We can using K matrix to put the poles at desired location and therefore change the respose of the system. 
+
+
+## How to chose K ?  
+Since we can change the respoce of the system by poles placement. We might ask ourself where should I put the poles, in other words, how to chose the K ? One idea is to construcet the problem into optimization framework that minimize the state error and control nameed LQR.
```
Comment (jskzhou) [inline bite-size-notes/week-4-d-Example of LQR-name.md]: This section makes it seem like LQR is a way of doing pole placement. It isn't clear to me that pole placement is another way of saying controller gain design (or vice versa). 

I think it might make more sense to highlight the differences between LQR and pole placement, rather than drawing equivalences between them.

File: bite-size-notes/week-4-d-Example of LQR-name.md [hunk 6]
```diff
@@ -0,0 +1,350 @@
...
+
+In class, we already disccused, by using HJB, the optimal control input u^* is a function of A,B,R,Q,P where
```
Comment (jskzhou) [inline bite-size-notes/week-4-d-Example of LQR-name.md:57]: HJB has not been mentioned up to now, it would help with clarity to define abbreviations before using them.

File: bite-size-notes/week-4-d-Example of LQR-name.md [hunk 7]
```diff
@@ -0,0 +1,350 @@
...
+\begin{aligned}
+	A \text{ is the A matrix of the system (know)};\\
+	B \text{ is the B matrix of the system (know)};\\
+	Q \text{ is a states weighing function (know)};\\
+	R \text{ is a control weighing function  (know)};\\
+	P \text{ is a matrix variable form the Riccati equation (Uknow)};
+ \end{aligned}
+$$
+
+And by solving the Ricatti equation, we can find the optimal control input $u^*$
```
Comment (jskzhou) [inline bite-size-notes/week-4-d-Example of LQR-name.md]: It's unclear here how the Riccati equation relates to LQR exactly, and why solving it yields the optimal control input. A sentence or two explaining this would be nice, since you have the nice table below so you don't have to explicitly derive everything.

File: bite-size-notes/week-4-d-Example of LQR-name.md [hunk 8]
```diff
@@ -0,0 +1,350 @@
...
+| Discret time finite horizon   | $$\sum_{n=1}^{\infty} x_{k}^T Q_k x_{k} + u_{k}^T R_k u_{k}\$$ | $P_k = Q_k +A^T P_{k+1}A-(A^TP_{k+1}B)(R_k+B^TP_{k+1}B)^{-1}(B^TP_{k+1}A)$ | 
+| Continuous time infinite horizon   | $\min \int_{0}^{\inf}x(t)^T Q(t) x(t) + u(t)^T R(t) u(t)$ |$PA+A^TP+Q-PBR^{-1}B^TP=0$ |
+| Discret time infinite horizon      | $$\min x_{k+1}^T Q_T x_{k+1} + \sum_{k=0}^k x_{k}^T Q_k x_{k} + u_{k}^T R_k u_{k}$$ | $P = Q +A^TPA-(A^TPB)(R+B^TPB)^{-1}(B^TPA)$| 
+
+|                                    |  u          |                     
+| ---------------------------------  | -------------------------------- |
+| Continuous time finite horizon     | $u^*(t) =-R^{-1}B^TP(t)x$ |
+| Discret time finite horizon        | $u^*_k =-(R _{k+1}+B^TP _{k+1}B^{-1}BP _{k+1}A) X _k$  |
+| Continuous time infinite horizon   | $u^* =-R^{-1}B^TPx$|
+| Discret time infinite horizon      | $u^*_k =-(R+B^TPB^{-1}(B^TPA)x$ |
```
Comment (jskzhou) [inline bite-size-notes/week-4-d-Example of LQR-name.md:84]: These tables are nice and lays out all the variations of LQR and Riccati equation for different dynamical system representations and forms and shows how the formulation can change. It's helpful that you included forms that we did not derive in class as well. 

The ordering of the table is a little confusing for me, since it separates the continuous/discrete time cases and also separates the time finite/infinite horizon cases. Changing the order may more intuitively show how the solution differs based on the problem formulation.

I also believe that the discrete time infinite horizon solution should be $u^*_k =-(R+B^TPB)^{-1}(B^TPA)x$

File: bite-size-notes/week-4-d-Example of LQR-name.md [hunk 9]
```diff
@@ -0,0 +1,350 @@
...
+ \end{aligned}
+$$
+
+And by solving the Ricatti equation, we can find the optimal control input $u^*$
+The variation of the optimal representies of LQR and correponding Riccati equation and control u are summirized below.  
+
+|                                    | LQR form                         | Riccati Equation                |                                
+| ---------------------------------  | -------------------------------- |-------------------------------- |
+| Continuous time finite horizon     | $\min \int_{0}^{T}x(t)^T Q(t) x(t) + u(t)^T R(t) u(t) + x(T)^T Q(T) x(T)$ | $0 =\dot{P}(t) + P(t)A+A^TP(t)+Q-P(t)BR^{-1}B^TP(t)$ | 
+| Discret time finite horizon   | $$\sum_{n=1}^{\infty} x_{k}^T Q_k x_{k} + u_{k}^T R_k u_{k}\$$ | $P_k = Q_k +A^T P_{k+1}A-(A^TP_{k+1}B)(R_k+B^TP_{k+1}B)^{-1}(B^TP_{k+1}A)$ |
```
Comment (jskzhou) [inline bite-size-notes/week-4-d-Example of LQR-name.md]: This LQR form also needs $\min$. The column label "LQR form" is vague, since this is an optimization problem, it's more clear to clearly state that it is the cost function for which the optimization is performed.

Also it isn't immediately clear what matrix $P$ is since it isn't just a matrix variable from the Riccati equation as you mentioned above, but is directly related to matrix $Q$

File: bite-size-notes/week-4-d-Example of LQR-name.md [hunk 10]
```diff
@@ -0,0 +1,350 @@
...
+|                                    |  u          |                     
+| ---------------------------------  | -------------------------------- |
+| Continuous time finite horizon     | $u^*(t) =-R^{-1}B^TP(t)x$ |
+| Discret time finite horizon        | $u^*_k =-(R _{k+1}+B^TP _{k+1}B^{-1}BP _{k+1}A) X _k$  |
+| Continuous time infinite horizon   | $u^* =-R^{-1}B^TPx$|
+| Discret time infinite horizon      | $u^*_k =-(R+B^TPB^{-1}(B^TPA)x$ |
+
+
+
+Fron this point, we narrow down the optimal problem to solving a differentail matrix equation. And I want to show how to solving the Ricatti equation for the continuous finite time using matlab ode45.
```
Comment (jskzhou) [inline bite-size-notes/week-4-d-Example of LQR-name.md]: Refer to the equations above as to **which** differential equation you're solving, "a differential equation" is vague here

File: bite-size-notes/week-4-d-Example of LQR-name.md [hunk 11]
```diff
@@ -0,0 +1,350 @@
...
+
+
+Fron this point, we narrow down the optimal problem to solving a differentail matrix equation. And I want to show how to solving the Ricatti equation for the continuous finite time using matlab ode45. 
+ 
+# Main body 
+## Solving the Recatti equation using matlab ode45 (Method from AA550 Nonlinear optimal control by Prof. Santosh Divasia)
+
+Please refer to the attached code. 
+
+The optimal control problem in terns on LQR is:
```
Comment (jskzhou) [inline bite-size-notes/week-4-d-Example of LQR-name.md]: It would be helpful to note which case of LQR we are dealing with here, would add readability!

File: bite-size-notes/week-4-d-Example of LQR-name.md [hunk 12]
```diff
@@ -0,0 +1,350 @@
...
+$$
+\dot{P} (t) =  -P(t)A-A^TP(t)-Q-P(t)BR^{-1}B^TP(t))
+$$
+
+Optimal control input is :
+
+$$u^*(t) =-R^{-1}B^TP(t)x$$
+
+By applying boundery condition, P matrix at final time, P(tf), is Q(T). 
+Therefore we can sovle the P for $\dot{P}$, by solving the differential equation backward in time with initial condition $P(0) = Q(T)$.
```
Comment (jskzhou) [inline bite-size-notes/week-4-d-Example of LQR-name.md]: I don't understand what you mean by "solve P for $\dot{P}$" here. It would help to refer to the math you've written. You can add equation numbers and refer to them in your writing.

I mentioned this above, but if there is a sentence or two (not full derivation) on how the Riccati equation relates to/comes from the cost function, and how solving it yields the optimal control input, it could add a lot of clarity to the math.

Also, some of your matrix variables are in plain text and some are in math mode. It would help with presentation if they are all in math mode!

File: bite-size-notes/week-4-d-Example of LQR-name.md [hunk 13]
```diff
@@ -0,0 +1,350 @@
...
+
+The command in matlab is: \
+
+[t,PV]=ode45(@(t,p) dpp(t,p,A,B,Q,R,E,F),[0 (tf-ti)],Ptf,options);
```
Comment (jskzhou) [inline bite-size-notes/week-4-d-Example of LQR-name.md:125]: Check the markdown formatting, this did not get rendered correctly

File: bite-size-notes/week-4-d-Example of LQR-name.md [hunk 14]
```diff
@@ -0,0 +1,350 @@
...
+
+
+The first step is rewrite the Riccati equation as in reversive form: 
+
+$$\dot{P}(t) =  P(t)A+A^TP(t)+Q-P(t)BR^{-1}B^TP(t)$$
```
Comment (jskzhou) [inline bite-size-notes/week-4-d-Example of LQR-name.md:130]: I like this approach to writing the solving process explicitly instead of using a software package!
It would be helpful to say what you mean by reverse form though (i.e. what is being reversed), it may be unclear what the reversed version of an equation is

File: bite-size-notes/week-4-d-Example of LQR-name.md [hunk 15]
```diff
@@ -0,0 +1,350 @@
...
+$$
+
+The Riccati equtaion is :
+
+$$
+0 =  \dot{P} (t) + (P(t)A+A^TP(t)+Q-P(t)BR^{-1}B^TP(t))
+$$
+
+$$
+\dot{P} (t) =  -P(t)A-A^TP(t)-Q-P(t)BR^{-1}B^TP(t))
```
Comment (jskzhou) [inline bite-size-notes/week-4-d-Example of LQR-name.md]: Extra bracket at the end of equation

File: bite-size-notes/week-4-d-Example of LQR-name.md [hunk 16]
```diff
@@ -0,0 +1,350 @@
...
+
+The first step is rewrite the Riccati equation as in reversive form: 
+
+$$\dot{P}(t) =  P(t)A+A^TP(t)+Q-P(t)BR^{-1}B^TP(t)$$
+
+To define a function named "dpp" as the reversed Riccati equation, which will be a anonymous function passing to ode45. 
+
+Since the inpout of the ode45 is a vactor, We need to reshape the vector to the original P matrix in the dpp founciton and reshape the output, $\dot{P}$ as a vector. 
+
+After solving the P, we will have Pm*Pm-by-n matrix. Pm is the dimention of the state or the roll number of the P matrix, n is the time stes number solving by ode45.
```
Comment (jskzhou) [inline bite-size-notes/week-4-d-Example of LQR-name.md]: Putting in-line math instead of plain text would improve the readability of this line
I am a little confused on what "roll number" is in this context, cannot find it on Google
Also, I can understand what $n$ is from reading the code, but this line leads me to believe that the value of $n$ is solved by ode45

File: bite-size-notes/week-4-d-Example of LQR-name.md [hunk 17]
```diff
@@ -0,0 +1,350 @@
...
+
+After solving the P, we will have Pm*Pm-by-n matrix. Pm is the dimention of the state or the roll number of the P matrix, n is the time stes number solving by ode45. 
+
+Because the solution is backward in time, we need to reverse the oder by using the matlab function fliup(). to reorder the PV vector and the corresponding time step in time. 
+
+Finally, we compute the optimal control u^*(t), at the time t = tm, by extracting the PV vector in corresponding row at tm into the original matrix P(tm) and plug it into the equation
+
+$u^*(tm) = R^{-1}BP(tm)$
+
+from t = 0 to t = final. The result is the series of the optimal control $u^*$.
```
Comment (jskzhou) [inline bite-size-notes/week-4-d-Example of LQR-name.md:144]: Your explanation of the code implementation is quite detailed and very helpful when looking at the code in tandem. Using more math mode in-line can make it easier to read and understand. Also, you can put code snippets below text to show what you mean, so the reader wouldn't need to scroll back and forth between the code and this section when reading

File: bite-size-notes/week-4-d-Example of LQR-name.md [hunk 18]
```diff
@@ -0,0 +1,350 @@
...
+$u^*(tm) = R^{-1}BP(tm)$
+
+from t = 0 to t = final. The result is the series of the optimal control $u^*$. 
+
+
+Let's do a example:
+The optimal control problem in terns on LQR is:
+
+$$
+	$\min \int_{0}^{1}x(t)^T Q(t) x(t) + u(t)^T R(t) u(t) + x(T)^T Q(T) x(T)$
```
Comment (jskzhou) [inline bite-size-notes/week-4-d-Example of LQR-name.md]: There's a few extra $ and I think a missing newline between the cost function and the constraint

File: bite-size-notes/week-4-d-Example of LQR-name.md [hunk 19]
```diff
@@ -0,0 +1,350 @@
...
+Where
+
+$$
+A =  \begin{bmatrix} 0 & 1  \\\ 3 & 0 \end{bmatrix} \\
+B =  \begin{bmatrix} 0  \\\ 3  \end{bmatrix} \\
+Q =  \begin{bmatrix} 0  \\\ 3  \end{bmatrix} \\
+R= 1 \\;
+Q(T) = \begin{bmatrix} 50 & 0  \\\ 0 & 50  \end{bmatrix} \\
+x = \begin{bmatrix} x1  \\\ x2 \end{bmatrix} \\
+$$
```
Comment (jskzhou) [inline bite-size-notes/week-4-d-Example of LQR-name.md:169]: Check the formatting here, the newlines don't appear to have rendered correctly.

File: bite-size-notes/week-4-d-Example of LQR-name.md [hunk 20]
```diff
@@ -0,0 +1,350 @@
...
+
+$$
+\text{u is one dimention.} \\
+$$
+
+$$
+\text{initial state} x(0) = \begin{bmatrix} 1  \\\ 1 \end{bmatrix} \\
+$$
+
+##code
```
Comment (jskzhou) [inline bite-size-notes/week-4-d-Example of LQR-name.md:179]: The code is great and super helpful, especially if someone wants to put it in a code sandbox and test it. I like that you commented each block and most of the function calls to make it clear what the code is doing.

File: bite-size-notes/week-4-d-Example of LQR-name.md [hunk 21]
```diff
@@ -0,0 +1,350 @@
...
+end
+```
+
+# Results: 
+First we can we plot the states without control, and find the system blows up, since the eigenvalue of the A matrix is on the RHP.
+![uncon](https://github.com/p8410077/AA548-spr2024/assets/11802603/b7168051-1b6c-4784-8c36-71f37404da7b)
+
+By inplementing LQR, the system converge to 0 in 1 second. 
+![image](https://github.com/p8410077/AA548-spr2024/assets/11802603/389fc3c2-0a2a-4066-815f-6815c59bd34f)
+
```
Comment (jskzhou) [inline bite-size-notes/week-4-d-Example of LQR-name.md:350]: The result plots are helpful. They can be formatted better with the titles added and legends resized and contained within the plot area. Also, you can save as PNG in MATLAB instead of screenshotting, while would give a better aspect ratio on these figures. 

Overlaying the plots would show really effectively how the LQR controlled system differs from the uncontrolled system. Showing the cost converging can also be a really helpful plot.

File: bite-size-notes/week-4-d-Example of LQR-name.md [hunk 22]
```diff
@@ -0,0 +1,350 @@
...
+end
+
+function dx = dyn(t,x,A,B)
+
+dx = A*x; 
+end
+```
+
+# Results: 
+First we can we plot the states without control, and find the system blows up, since the eigenvalue of the A matrix is on the RHP.
```
Comment (jskzhou) [inline bite-size-notes/week-4-d-Example of LQR-name.md]: It may be unclear to the reader what you mean by "the system blows up" in the context of self-contained notes, can say the system diverges to infinity. It isn't clear from the above what the eigenvalues of $A$ are or that it is on the RHP, supporting math would be useful. 

Also, as the above is in time domain, referring to the right-half of the complex plane seems unrelated to the above analysis.

File: bite-size-notes/week-4-d-Example of LQR-name.md [hunk 23]
```diff
@@ -0,0 +1,350 @@
...
+
+$$
+	\dot{x}(t) = Ax(t)+ Bu(t) \\
+$$
+
+$$
+	y(t) = Cx(t) +Du(t) \
+$$
+
+By finding the location of the poles of matrix A, we can understand if the system is stable. We can design a controller that stabilize the system, if the system is unstable, or drive the system to converge faster, if the system is stable.
```
Comment (jskzhou) [inline bite-size-notes/week-4-d-Example of LQR-name.md:25]: The singularity of "location" and plurality of "poles" implies that the poles of $A$ are all the same.

File: bite-size-notes/week-4-d-Example of LQR-name.md [hunk 24]
```diff
@@ -0,0 +1,350 @@
...
+# Results: 
+First we can we plot the states without control, and find the system blows up, since the eigenvalue of the A matrix is on the RHP.
+![uncon](https://github.com/p8410077/AA548-spr2024/assets/11802603/b7168051-1b6c-4784-8c36-71f37404da7b)
+
+By inplementing LQR, the system converge to 0 in 1 second. 
+![image](https://github.com/p8410077/AA548-spr2024/assets/11802603/389fc3c2-0a2a-4066-815f-6815c59bd34f)
+
+# Conclusion: 
+1. We succesfully use ode45 to solve Ricatti equation and find the optimal control for continuous finite time problem.
+2. For other variation of the LQR need different technic to solve, for example: the infinit time problen is not a ODE, therefore can't use ode45.
```
Comment (jskzhou) [inline bite-size-notes/week-4-d-Example of LQR-name.md]: It may be helpful to write out the math for the infinite time horizon problem and show why it isn't an ODE, and maybe a line on what other approaches there can be approach such problems.

**Comment by jskzhou** [comment]
Overall I think there is a lot of good math and details in your notes, and can be really helpful for anyone wanting to review LQR and would like to see an example/implementation and the corresponding results. Providing justification to some statements and verifying some of the claims, as well as polishing the presentation of the notes, would help greatly in complementing your math and code, as well as be more useable and clear to the reader.

File: bite-size-notes/week-4-d-Example of LQR-name.md [hunk 25]
```diff
@@ -0,0 +1,350 @@
...
+| ---------------------------------  | -------------------------------- |-------------------------------- |
+| Continuous time finite horizon     | $\min \int_{0}^{T}x(t)^T Q(t) x(t) + u(t)^T R(t) u(t) + x(T)^T Q(T) x(T)$ | $0 =\dot{P}(t) + P(t)A+A^TP(t)+Q-P(t)BR^{-1}B^TP(t)$ | 
+| Discret time finite horizon   | $$\sum_{n=1}^{\infty} x_{k}^T Q_k x_{k} + u_{k}^T R_k u_{k}\$$ | $P_k = Q_k +A^T P_{k+1}A-(A^TP_{k+1}B)(R_k+B^TP_{k+1}B)^{-1}(B^TP_{k+1}A)$ | 
+| Continuous time infinite horizon   | $\min \int_{0}^{\inf}x(t)^T Q(t) x(t) + u(t)^T R(t) u(t)$ |$PA+A^TP+Q-PBR^{-1}B^TP=0$ |
+| Discret time infinite horizon      | $$\min x_{k+1}^T Q_T x_{k+1} + \sum_{k=0}^k x_{k}^T Q_k x_{k} + u_{k}^T R_k u_{k}$$ | $P = Q +A^TPA-(A^TPB)(R+B^TPB)^{-1}(B^TPA)$| 
+
+|                                    |  u          |                     
+| ---------------------------------  | -------------------------------- |
+| Continuous time finite horizon     | $u^*(t) =-R^{-1}B^TP(t)x$ |
+| Discret time finite horizon        | $u^*_k =-(R _{k+1}+B^TP _{k+1}B^{-1}BP _{k+1}A) X _k$  |
```
Comment (jskzhou) [inline bite-size-notes/week-4-d-Example of LQR-name.md]: I believe the discrete time finite horizon expression should be $u^*_k =-(R _{k+1}+B^TP _{k+1}B)^{-1}(BP _{k+1}A) x _k$

File: bite-size-notes/week-4-d-Example of LQR-name.md (hunk 1 above)
Comment (p8410077) [reply bite-size-notes/week-4-d-Example of LQR-name.md]: You are right. The statements is too strong.  I make a new statement.

File: bite-size-notes/week-4-d-Example of LQR-name.md (hunk 2 above)
Comment (p8410077) [reply bite-size-notes/week-4-d-Example of LQR-name.md]: I specified the stander in this note means system without reference input

File: bite-size-notes/week-4-d-Example of LQR-name.md (hunk 3 above)
Comment (p8410077) [reply bite-size-notes/week-4-d-Example of LQR-name.md]: Thanks. I made a clarification

File: bite-size-notes/week-4-d-Example of LQR-name.md (hunk 8 above)
Comment (p8410077) [reply bite-size-notes/week-4-d-Example of LQR-name.md:84]: Thanks for correct the equation.

File: bite-size-notes/week-4-d-Example of LQR-name.md (hunk 9 above)
Comment (p8410077) [reply bite-size-notes/week-4-d-Example of LQR-name.md]: Thanks for bringing up some misleading content. I correct them.

File: bite-size-notes/week-4-d-Example of LQR-name.md (hunk 10 above)
Comment (p8410077) [reply bite-size-notes/week-4-d-Example of LQR-name.md]: Thanks, I specified the what is the equation.

File: bite-size-notes/week-4-d-Example of LQR-name.md (hunk 11 above)
Comment (p8410077) [reply bite-size-notes/week-4-d-Example of LQR-name.md]: Thanks. I specified the case.

File: bite-size-notes/week-4-d-Example of LQR-name.md (hunk 12 above)
Comment (p8410077) [reply bite-size-notes/week-4-d-Example of LQR-name.md]: I make all matrix in math mode, and specified the equation I am solving

File: bite-size-notes/week-4-d-Example of LQR-name.md (hunk 13 above)
Comment (p8410077) [reply bite-size-notes/week-4-d-Example of LQR-name.md:125]: Got it

File: bite-size-notes/week-4-d-Example of LQR-name.md (hunk 15 above)
Comment (p8410077) [reply bite-size-notes/week-4-d-Example of LQR-name.md]: got it

File: bite-size-notes/week-4-d-Example of LQR-name.md (hunk 18 above)
Comment (p8410077) [reply bite-size-notes/week-4-d-Example of LQR-name.md]: got  it

File: bite-size-notes/week-4-d-Example of LQR-name.md (hunk 19 above)
Comment (p8410077) [reply bite-size-notes/week-4-d-Example of LQR-name.md:169]: got it

File: bite-size-notes/week-4-d-Example of LQR-name.md (hunk 20 above)
Comment (p8410077) [reply bite-size-notes/week-4-d-Example of LQR-name.md:179]: thanks

File: bite-size-notes/week-4-d-Example of LQR-name.md (hunk 4 above)
Comment (p8410077) [reply bite-size-notes/week-4-d-Example of LQR-name.md]: I put the K gain on the bottom, but did not add the reference input, it might be reasonable to include reference, but the resource I refer did not specify there is reference input

File: bite-size-notes/week-4-d-Example of LQR-name.md (hunk 5 above)
Comment (p8410077) [reply bite-size-notes/week-4-d-Example of LQR-name.md]: I make a clarification in the revision.

File: bite-size-notes/week-4-d-Example of LQR-name.md (hunk 22 above)
Comment (p8410077) [reply bite-size-notes/week-4-d-Example of LQR-name.md]: Because eigenvalue of A is on the RHP, so the system is unstable and state goes to infinity

File: bite-size-notes/week-4-d-Example of LQR-name.md (hunk 25 above)
Comment (p8410077) [reply bite-size-notes/week-4-d-Example of LQR-name.md]: Thanks for correct the equation, I'll fix it

File: bite-size-notes/week-4-d-Example of LQR-name.md (hunk 24 above)
Comment (p8410077) [reply bite-size-notes/week-4-d-Example of LQR-name.md]: Thanks, I specified that ode45 can't be use for discrete time equation

File: bite-size-notes/week-4-d-Example of LQR-name.md (hunk 22 above)
Comment (jskzhou) [reply bite-size-notes/week-4-d-Example of LQR-name.md]: yes, my point was just that eigenvalues is not super intuitive in time domain analysis, so it may be helpful to write out explicitly what the values of those eigenvalues are because you have a good numerical example example!
